python/ray/worker.py

```diff
@@ -849,8 +849,11 @@ def shutdown(exiting_interpreter=False):
 def sigterm_handler(signum, frame):
     sys.exit(signal.SIGTERM)
 
-
-signal.signal(signal.SIGTERM, sigterm_handler)
+try:
+    signal.signal(signal.SIGTERM, sigterm_handler)
+except ValueError:
+    logger.warning("Processes might not be terminated properly if SIGTERM is sent"
+                   "(tried to set signal from outside the main thread).")
```

Comment:
is this the only scenario that this could happen? if not, leaving a generic error message is fine:
"Failed to set SIGTERM handler, processes might not be cleaned up properly on exit"